clippy_lints/src/undocumented_unsafe_blocks.rs

```diff
@@ -440,26 +518,33 @@ enum HasSafetyComment {
 
 /// Checks if the lines immediately preceding the item contain a safety comment.
 #[allow(clippy::collapsible_match)]
-fn item_has_safety_comment(cx: &LateContext<'_>, item: &hir::Item<'_>) -> HasSafetyComment {
-    match span_from_macro_expansion_has_safety_comment(cx, item.span) {
+fn item_has_safety_comment(cx: &LateContext<'_>, hir_id: HirId, owner: hir::OwnerNode<'_>) -> HasSafetyComment {
+    match span_from_macro_expansion_has_safety_comment(cx, owner.span()) {
         HasSafetyComment::Maybe => (),
         has_safety_comment => return has_safety_comment,
     }
 
-    if item.span.ctxt() != SyntaxContext::root() {
+    if owner.span().ctxt() != SyntaxContext::root() {
```

Comment:
Another nit while we're in the area

```suggestion
    if owner.span().from_expansion() {
```